### PR TITLE
18usa privates batch4

### PR DESCRIPTION
**Pull request by ventusignis**
Adds the two train running shenanigans privates to the game.
This will make it 18/30 done
This also adds the pullman train and the ability to add it to a train as an ability (like the two new privates)

### CODE REVIEW

File: lib/engine/graph.rb [hunk 1]
```diff
@@ -23,7 +24,7 @@ def clear
       @connected_hexes.clear
       @connected_nodes.clear
       @connected_paths.clear
-      @reachable_hexes.clear
```
Comment (tobymao) [inline lib/engine/graph.rb:26]: why did you delete this

Comment (tobymao) [reply lib/engine/graph.rb:26]: can you undo these changes and just use a new graph like 1873?

Comment (ventusignis) [reply lib/engine/graph.rb]: I've undid the changes and made a separate graph for this case

**Review by tobymao — APPROVED** [review]
